### PR TITLE
Add snail

### PR DESCRIPTION
**Pull request by glossopm**

### CODE REVIEW

**Review by infinitydeep — APPROVED** [review]
tests pass locally and through circleci, tests have value and show validation works.

**Review by MikeSilverstone — DISMISSED** [review]

File: routes/races/add_snail_to_race.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        if validate_snail_in_same_race(race_id, form.snail_id.data):
+            flash_redirect("This snail is already in the selected race", "add_snail_to_race.add_snail_to_race", race_id, round_id)
+
+        elif validate_snail_in_same_round(round_id, race_id, form.snail_id.data):
+            flash_redirect("This snail is already racing in the selected round", "add_snail_to_race.add_snail_to_race", race_id, round_id)
+
+        elif not validate_snail_in_inflight_round(round_id):
+            flash_redirect("This round in ineligible for snails to be added, please check the times and try again", "add_snail_to_race.add_snail_to_race", race_id, round_id)
+
+        else:
```
[the remark this answers is not in the document]
Comment (glossopm) [reply routes/races/add_snail_to_race.py]: Keeping the `else` statement for readability.

File: routes/races/add_snail_to_race.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    for race in races_in_round:
+        race_participants = RaceParticipants().get_race_participants_race_id(race.id)
+        for snail in race_participants:
+            if int(snail.id_snail) == int(snail_id):
+                i = True
+        return i
+
+
+def validate_snail_in_inflight_round(round_id):
+    future_rounds = Round().get_future_round_times()
```
[the remark this answers is not in the document]
Comment (glossopm) [reply routes/races/add_snail_to_race.py]: I'll speak with the person who wrote the tests to include these in their revision.

[the remark this answers is not in the document]
Comment (glossopm) [reply routes/races/add_snail_to_race.py]: These changes have been complete.

**Pull request merged**